### ISSUE TITLE
PHP Installation doesn't work without sleep

### ISSUE THREAD

**Issue by Zoiiid** [nextcloud/vm#2358]
Dear Nextcloud-Team

I wanted to install your service on a Proxmox Container instance via your installation Script that you've kindly provided.

Now i've tried to install it but it kept on hanging during the PHP installation, more specifically during the "restart_webserver" function on line 456. https://github.com/nextcloud/vm/blob/f45a3d29d90573dbdce4ebf8605d3696b6a9ebb9/nextcloud_install_production.sh#L456

So what i did to mitigate this issue: I put a "sleep 5" command before the restarting of the webserver, so that PHP will install cleanly.

Could you please implement this on your master script, so that people won't get this issue in the future?

Thanks and have a nice day
Zoiiid

**Comment by enoch85** [comment]
This must be a very specifik use case, worked every time for the 10 000+ installationa done so far.

Will add it anyway though to compensate for slow disks or whatever reason that might have caused it.

A PR would be welcome! ;)

**Comment by michaelbrown2** [comment]
I also ran into this issue. I'm running it in a Virtual Machine under Proxmox with a fresh install of Ubuntu 22.04.  4 cores, 4GB of ram, one VirtIO 48GB Hard Drive running off an SSD, one VirtIO 2TB Hard Drive running off a zfs pool of two spinning 8TB drives in mirror. 

The odd thing was, the first time I ran the script I forgot to add the 2TB virtual disk, and the script ran fine.  I finished it, and then realized I'd rather have the script provision the data drive, so i wiped it clean and reran install with the disk attached.

At the point in the script, it said: 
systemctl restart php8.1-fpm.service failed

(sorry, I didn't copy the rest, something about coming here to make a bug report) and had me run journalctl -xue php8.1-fpm.service , the only log that was relevent said it had tried to restart too quickly or too many times (again, I cant remember and I finally got it running, I don't want to go through this again...)

Adding the sleep 5 before line 456 worked for me.  I don't know why it's targeting so few installations.  I'm hoping this information might be helpful for you all to track it down though!

**Comment by enoch85** [comment]
OK, so issue here seems to be that you re ran thee scripts. That's not supported.

**Comment by michaelbrown2** [comment]
> OK, so issue here seems to be that you re ran the scripts. That's not supported.

I’m sorry, I failed to explain that correctly. Each time the script failed, I scrubbed the Virtual Machine and reinstalled Ubuntu 22.04 Server. I then did a full apt update and apt upgrade, and then ran the script. I did not rerun the script after a failed attempt. I did TRY, but the script let me know that I couldn’t.

I did this 5 times before finally giving up and trying to research the error. Google returned nothing, but then I found this Issue and tried the solution and it worked. I just wanted to hopefully help y’all identify the root issue and make the script a little more flexible.
